packages/api/src/services/emp-stats.ts

```diff
@@ -34,51 +33,76 @@ export default (config: Config, appState: Dependencies) => {
     return getOrCreateHistoryTable(address).create(data);
   }
 
-  async function updateHistory(address: string) {
-    const stat = await getLatestStats(address);
-    assert(uma.utils.exists(stat.timestamp), "stats require timestamp");
-    if (await hasStatsHistory(address, stat.timestamp)) return stat;
-    const update = {
-      address: stat.address,
-      tvl: stat.tvl,
-      tvm: stat.tvm,
-      timestamp: stat.timestamp,
-    };
-    return createStatHistory(address, update);
+  async function updateHistory(empAddress: string) {
+    const stat = await getLatestStats(empAddress);
+    assert(uma.utils.exists(stat.timestamp), "stats require timestamp for emp: " + empAddress);
+    if (await hasStatsHistory(empAddress, stat.timestamp)) return stat;
+    // remove id from the latest stats, since this gets overriden with history stats by timestamp
+    const { id, ...historyStat } = stat;
+    return createStatHistory(empAddress, historyStat as EmpHistoryStat);
```

Comment:
this is really tricky, had issues getting this to work. Heres what lead to this: 
There are 2 data structures, one that represents the latest stats, the other represents the historical stat. Timestamp is optional on the first, but required on the second ( address is key in the latest table but timestamp is the id in historical table).  For some reason, typescript doesnt care that i validate timestamp exists, it still complains that its the wrong type, so basically i have to validate and then cast, telling typescript its now the right type.  Heres an example of it failing:

![image](https://user-images.githubusercontent.com/4429761/123846380-4c0c9180-d8e3-11eb-9f75-f9c876763662.png)

somehow it still thinks timestamp can be undefined:
![image](https://user-images.githubusercontent.com/4429761/123846431-5b8bda80-d8e3-11eb-916b-fcdad73c399d.png)

Reply:
I think this was mainly to make it easy to construct the latest stats object partially, given only an address.  Becuase we have invalid EMPs we might not be able to get any "stats" for those addresses (including timestamp). 

I might be able to make timestamp required but at this point though might be moot since i am getting rid of this object in a future PR.